### ISSUE TITLE
sbt mysubproject/run doesn't work

### ISSUE THREAD

**Issue by jroper** [playframework/playframework#3484]
If trying to run Play dev mode on a sub project using `sbt mysubproject/run`, it fails with the following exception when you try to use the app:

```
[ERROR] [2014-10-07 13:40:03,019] [application] [New I/O worker #1] -

! Internal server error, for (GET) [/login] ->

java.util.NoSuchElementException: None.get
at scala.None$.get(Option.scala:313) ~[scala-library-2.11.2.jar:na]
at scala.None$.get(Option.scala:311) ~[scala-library-2.11.2.jar:na]
at play.PlayReloader$$anon$1.reload(PlayReloader.scala:88) ~[na:na]
at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:122) ~[play_2.11-2.3.4.jar:2.3.4]
at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:120) ~[play_2.11-2.3.4.jar:2.3.4]
at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.forkjoin.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1361) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) ~[scala-library-2.11.2.jar:na]
at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) ~[scala-library-2.11.2.jar:na]
[WARN] [2014-10-07 13:40:03,097] [play] [New I/O worker #1] - No application found at invoker init
[INFO] [2014-10-07 13:40:03,352] [akka.event.slf4j.Slf4jLogger] [play-akka.actor.default-dispatcher-2] - Slf4jLogger started
```

It seems that we're not executing the `playReload` task for the right project, it's probably being run for the root project, not the selected sub project.

**Comment by pauldraper** [comment]
Are these the same problem (`mysubproject/run` or `use mysubproject; run`) or different ones?

I'd love to see this fixed.

I'm trying to figure out `playReload` and scopes, etc., but I'm new to SBT. If someone knows anything that could help, let me know.

**Comment by pauldraper** [comment]
@marekzebrowski, I too had this problem with `project mysubproject`.

I was working on combining separate SBT projects into one project with subprojects.

Then I realized that in my mess of .scala and .sbt files, I was actually downloading old artifacts for the projects. I removed the old .sbt files (I had one in each project), and it worked. At least, I think that's what happened.

---

**tl;dr**

Using `project mysubproject` and `run`, rather than `mysubproject/run`, works.

**Comment by romusz** [comment]
the workaround works for me as well. thanks.

**Comment by ms-tg** [comment]
@jroper I'm not sure this workaround is sufficient for us. It prevents us from scripting the launch of Play web-apps, which we want to do in various scenarios, as it requires interacting with the console.

Do you have a further workaround for the scripting case?

/cc @jaramir

**Comment by flicken** [comment]
@ms-tg Is it sufficient to pass multiple sbt commands via the command line?  For example:

    sbt ';project mysubproject ; run'

**Comment by jroper** [comment]
See above for fix, are you on 2.3.x?  We should have a 2.3.x release with this fixed soon.

**Comment by ms-tg** [comment]
Ah excellent @jroper thank you. We are indeed on 2.3.x for the affected projects, so thank you, we will look for the release with the fix.

Separately, just within TIM Group, @jaramir can get back to @flicken on wither the "quotes and semicolon" workaround is sufficient for our uses?

**Comment by schmitch** [comment]
Hello, seems like this issue reappeared for 2.4.x
